src/main/java/com/bookstore/repository/book/BookSpecificationProviderManager.java

```diff
@@ -0,0 +1,25 @@
+package com.bookstore.repository.book;
+
+import com.bookstore.model.Book;
+import com.bookstore.repository.SpecificationProvider;
+import com.bookstore.repository.SpecificationProviderManager;
+import java.util.List;
+import lombok.RequiredArgsConstructor;
+import org.springframework.stereotype.Component;
+
+@Component
+@RequiredArgsConstructor
+public class BookSpecificationProviderManager implements SpecificationProviderManager<Book> {
+
+    private final List<SpecificationProvider<Book>> bookSpecificationProviders;
+
+    @Override
+    public SpecificationProvider<Book> getSpecificationProvider(String key) {
+
+        return bookSpecificationProviders.stream()
+                .filter(sp -> sp.getKey().equals(key))
+                .findFirst()
+                .orElseThrow(() -> new RuntimeException("Can't find correct specification "
```

Comment:
```suggestion
                .filter(provider-> provider.getKey().equals(key))
                .findFirst()
                .orElseThrow(() -> new NoSuchElementException("Can't find correct specification "
```
better to throw more specific exceptions where possible, and to avoid short naming in streams